tests/Feature/Repositories/AccountRepositoryTest.php

```diff
@@ -39,6 +42,28 @@ public function setRepository()
         $this->repo = resolve(AccountRepositoryInterface::class);
     }
 
+    /**
+     * Make a new User object.
+     *
+     * @param array $userProperties
+     * @param array $accountProperties
+     * @return \App\User
+     */
+    public function makeUser(array $userProperties = [], array $accountProperties = [])
```

Comment:
I'm wondering if we can abstract this to the `RepositoryCrudTestCase` and pass in `$this->user` to make this available in all repository tests ?

Can we see multiple tests needing access to similar logic ?
Can we see creating maybe a Trait `use RepositoryTestFactories;` (or something similar) ?

Reply:
Yes, definitely. I'm confident that anything useful in this respect will present itself as I go through the tests within the scope of this PR. I'll identify those traits and pencil them in as I go.

Reply:
Through further investigation, maybe its better to just *NOT* try to abstract this logic.

It becomes more complicated then what its worth when there isn't any real reason to not just do it this way.

This is the solution that i came up with *without a DTO object*

**RepositoryReadOnlyTestCase**
```php
/**
     * Make a collection of Model instances
     *
     * @param array $classes
     * @return array
     */
    public function make(array $classes)
    {
        $models = collect([]);

        foreach ($classes as $class => $properties) {
            $models->push(factory($class)->make($properties));
        }

        return $models;
    }
```

**AccountRepositoryTest**
```php
/**
     * Make a new User object.
     *
     * @param array $userProperties
     * @param array $accountProperties
     * @return \App\User
     */
    public function makeUser(array $userProperties = [], array $accountProperties = [])
    {
        $models = $this->make([
            $this->user->class() => $userProperties,
            $this->repo->class() => $accountProperties
        ]);

        $user = $models[$this->user->class()];
        $account = $models[$this->repo->class()];

        $user = $this->user->create([
            'username' => $user->username,
            'password' => $user->password,
            'account' => $account->toArray(),
        ]);

        return $user;
    }
```
As you can see here its not doing much for us. Even creating a DTO isn't going to make any difference.

Reply:
Okay, having read your review, I agree. I'll revert my previous commit and then go forward with the examples your provided in your previous reply.

Reply:
Previous reply ?

Reply:
Yeah, the one I was replying to, in which you proposed a solution: https://github.com/indiehd/web-api/pull/93#discussion_r219548704

Reply:
No No, i wasn't proposing that solution, that solution is what i was using as an example as to how much effort would be involved into abstracting this logic where you get the same results from the one-liners used below.

I am suggesting no abstraction, and suggesting when you need to "make" a new model instance simply just call the `factory...->make()` method as you have been doing.

Reply:
Ahhh, I'm with you now. Sorry... I was focusing on the rationale and not the actual code. Understood.